trento-support.sh

```diff
@@ -19,232 +19,234 @@ OUTPUT=/dev/stdout
-            done
-            ;;
-
-        r)
-            RELEASE_NAME=$OPTARG
-            ;;
-
-        n)
-            NAMESPACE=$OPTARG
-            ;;
-
-        *)
-            usage
-            exit 0
-            ;;
-        esac
-    done
-
-    return 0
+	# abort if we have less than 2 arguments
+	if [[ $# -lt 2 ]]; then
+		usage
+		exit 1
+	fi
+
+	check_deps
+
+	local arg=
+
+	for arg; do
+		local delim=""
+		case "$arg" in
+		--help) args="${args}-h " ;;
+		--output) args="${args}-o " ;;
+		--collect) args="${args}-c " ;;
+		--release-name) args="${args}-r " ;;
+		--namespace) args="${args}-n " ;;
+
+		*)
+			[[ "${arg:0:1}" == "-" ]] || delim="\""
+			args="${args}${delim}${arg}${delim} "
+			;;
+		esac
+	done
+
+	eval set -- "$args"
+
+	while getopts "ho:c:r:n:" OPTION; do
+		case $OPTION in
+		h)
+			usage
+			exit 0
+			;;
+
+		o)
+			OUTPUT_OPT="${OPTARG}"
+			if [[ ${OUTPUT_OPT} != "stdout" && ${OUTPUT_OPT} != "file" && ${OUTPUT_OPT} != "file-tgz" ]]; then
+				echo "Invalid output type: $OUTPUT_OPT"
+				usage
+				exit 1
+			fi
+
+			if [ "${OUTPUT_OPT}" = "stdout" ]; then
+				OUTPUT="/dev/stdout"
+			elif [ "${OUTPUT_OPT}" = "file-tgz" ]; then
+				OUTPUT="$(date +%Y-%m-%d_%H%M)_support.txt"
+				COMPRESS=true
+			else
+				OUTPUT="$(date +%Y-%m-%d_%H%M)_support.txt"
+			fi
+			;;
+
+		c)
+			COLLECT=$OPTARG
+			IFS=, read -r -a arr <<<"${COLLECT}"
+			for key in "${!arr[@]}"; do
+				if [[ -z "${VALID_FACILITIES[${arr[$key]}]}" ]]; then
+					printf '%s: unsupported facility\n' "${arr[$key]}"
+					usage
+					exit 1
+				fi
+			done
+			;;
+
+		r)
+			RELEASE_NAME=$OPTARG
+			;;
+
+		n)
+			NAMESPACE=$OPTARG
+			;;
+
+		*)
+			usage
+			exit 0
+			;;
+		esac
+	done
+
+	return 0
 }
 
 check_deps() {
-    if ! which jq >/dev/null 2>&1; then
-        echo "error: jq is required and not installed"
-        exit 1
-    fi
-    if ! which yq >/dev/null 2>&1; then
-        echo "error: yq is required and not installed"
-        exit 1
-    fi
-    if ! which helm >/dev/null 2>&1; then
-        echo "error: helm is required and not installed"
-        exit 1
-    fi
-    if ! which kubectl >/dev/null 2>&1; then
-        echo "error: kubectl is required and not installed"
-        exit 1
-    fi
-    if ! which crictl >/dev/null 2>&1; then
-        COLLECT_CRICTL=false
-    fi
-    if ! which k3s >/dev/null 2>&1; then
-        COLLECT_K3S=false
-    fi
+	if ! which jq >/dev/null 2>&1; then
+		echo "error: jq is required and not installed"
+		exit 1
+	fi
+	if ! which yq >/dev/null 2>&1; then
+		echo "error: yq is required and not installed"
+		exit 1
+	fi
+	if ! which helm >/dev/null 2>&1; then
+		echo "error: helm is required and not installed"
+		exit 1
+	fi
+	if ! which kubectl >/dev/null 2>&1; then
+		echo "error: kubectl is required and not installed"
+		exit 1
+	fi
+	if ! which crictl >/dev/null 2>&1; then
+		COLLECT_CRICTL=false
+	fi
+	if ! which k3s >/dev/null 2>&1; then
+		COLLECT_K3S=false
+	fi
 }
 
-check_deps
```

Comment:
This is the only _structural_ change introduced. Dependencies are now checked AFTER the command line is read. This allows us to show the helper even with some missing dependencies (the script would always act as broken otherwise)